zcash_client_sqlite/src/testing.rs

```diff
@@ -323,6 +319,48 @@ where
         self.cache.insert(&compact_block)
     }
 
+    pub(crate) fn initialize_from_chain_state(
+        &mut self,
+        state: ChainState,
+        prior_sapling_roots: &[CommitmentTreeRoot<sapling::Node>],
+        #[cfg(feature = "orchard")] prior_orchard_roots: &[CommitmentTreeRoot<MerkleHashOrchard>],
+    ) -> Result<(), ShardTreeError<commitment_tree::Error>> {
+        self.wallet_mut()
+            .put_sapling_subtree_roots(0, prior_sapling_roots)?;
+        #[cfg(feature = "orchard")]
+        self.wallet_mut()
+            .put_orchard_subtree_roots(0, prior_orchard_roots)?;
```

Comment:
Given that this method is named `initialize_from_chain_state`, it should either assert that the trees are empty, or clear them before adding anything. If neither of those should be done, then rename the method.